File: HealthCareBillingSystem/src/interfaces/bill.ts
export interface Bill {
  id: string;
  patientId: string;
  patientName: string;
  date: string;
  amount: number;
  tax: number;
  discount: number;
  totalAmount: number;
  status: 'PENDING' | 'PAID' | 'CANCELLED';
  paymentMethod?: 'CASH' | 'CARD' | 'INSURANCE' | 'ONLINE';
  paymentDate?: string;
  items: BillItem[];
  notes?: string;
  createdAt: string;
  updatedAt: string;
}

export interface BillItem {
  description: string;
  quantity: number;
  unitPrice: number;
  total: number;
  category: string;
}

export interface BillStats {
  totalRevenue: number;
  pendingBills: number;
  paidBills: number;
  averageBillAmount: number;
  monthlyRevenue: { month: string; revenue: number }[];
}

File: HealthCareBillingSystem/HealthCarePaymentTrackerAPi/HealthCareBilling.Model/Model/Bill.cs
using System;
using System.Collections.Generic;

namespace HealthCareBilling.Model
{
    public class Bill
    {
        public int Id { get; set; }
        public int PatientId { get; set; }
        public DateTime? BillDate { get; set; }
        public DateTime DueDate { get; set; }
        public decimal TotalAmount { get; set; }
        public string? Status { get; set; }
        public string? Description { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
        public List<BillItem> BillItems { get; set; } = new List<BillItem>();
    }

    public class BillItem
    {
        public int Id { get; set; }
        public int BillId { get; set; }
        public string Description { get; set; } = string.Empty;
        public int Quantity { get; set; }
        public decimal UnitPrice { get; set; }
        public decimal TotalAmount { get; set; }
    }
}